admin/commands/state_synchronization/execute_script.go

```diff
@@ -12,20 +13,41 @@ import (
 
 var _ commands.AdminCommand = (*ReadExecutionDataCommand)(nil)
 
+// scriptData holds the parsed input data for ExecuteScriptCommand.
 type scriptData struct {
 	height    uint64
 	script    []byte
 	arguments [][]byte
 }
 
+// ExecuteScriptCommand is an admin command that executes a Cadence script.
 type ExecuteScriptCommand struct {
-	scriptExecutor execution.ScriptExecutor
+	scriptExecutor      execution.ScriptExecutor
+	registersAsyncStore *execution.RegistersAsyncStore
 }
 
-func (e *ExecuteScriptCommand) Handler(ctx context.Context, req *admin.CommandRequest) (interface{}, error) {
+// Handler executes the Cadence script against the blockchain state at the
+// specified block height.
+//
+// Expected error returns during normal operation:
+//   - [version.ErrOutOfRange] - if incoming block height is higher that last handled block height.
+//   - [execution.ErrIncompatibleNodeVersion] - if the block height is not compatible with the node version.
+//   - [storage.ErrNotFound] - if data was not found.
+//   - [storage.ErrHeightNotIndexed] - if the requested height is below the first indexed height or above the latest indexed height.
+//   - [fvmerrors.ErrCodeScriptExecutionCancelledError] - if script execution canceled.
+//   - [fvmerrors.ErrCodeScriptExecutionTimedOutError] - if script execution timed out.
+//   - [fvmerrors.ErrCodeComputationLimitExceededError] - if script execution computation limit exceeded.
+//   - [fvmerrors.ErrCodeMemoryLimitExceededError] -if script execution memory limit exceeded.
+//   - [indexer.ErrIndexNotInitialized] - if data for block is not available.
```

Comment:
these are a lot of errors to handle. what logic is handling them? I wonder if we should use an admin sentinel error that groups them?